src/main/java/Duke.java

```diff
@@ -1,10 +1,245 @@
+            printAddedMessage();
+        } else if (isValidDeadline(input)) {
+            addToList(input, DEADLINE);
+            printAddedMessage();
+        } else if (isValidEvent(input)) {
+            addToList(input, EVENT);
+            printAddedMessage();
+        } else {
+            setError(true);
+        }
+        if (error) {
+            System.out.println(TAB + "Invalid input");
+            setError(false);
+        }
+        System.out.println(LINES);
+    }
+
+
+    /**
+     * This function adds the input description into list accordingly, based on its task type.
+     *
+     * @param input input given by the user.
+     * @param taskType task type identified beforehand (DEADLINE/EVENT/TODO).
+     */
+    private static void addToList(String input, String taskType) {
+        String[] parameters = new String[2];
+        getParameters(parameters, input, taskType);
+        switch (taskType) {
+        case TODO:
+            getParameters(parameters, input, TODO);
+            list[listIndexTracker] = new Todo(parameters[DESCRIPTION]);
+            break;
+        case DEADLINE:
+            getParameters(parameters, input, DEADLINE);
+            list[listIndexTracker] = new Deadline(parameters[DESCRIPTION], parameters[DATETIME]);
+            break;
+        case EVENT:
+            getParameters(parameters, input, EVENT);
+            list[listIndexTracker] = new Event(parameters[DESCRIPTION], parameters[DATETIME]);
+            break;
+        }
+
+        listIndexTracker++;
+
+    }
+
+    /**
+     * This function prints the message when task is added to list.
+     */
+    private static void printAddedMessage() {
+        System.out.println(TAB + "Got it. I've added this task: ");
+        System.out.println(TAB + TAB + list[listIndexTracker - 1]);
+        System.out.println(TAB + String.format("Now you have %d tasks in the list.", listIndexTracker));
+    }
+
+    /**
+     * This function prints the individual elements in list.
+     */
+    private static void printList() {
+        for (int i = 0; i < listIndexTracker; i++) {
+            System.out.format(TAB + "% 3d.", i + 1);
+            System.out.println(list[i]);
+        }
+    }
+
+    /**
+     * This function sets done to true, stopping while loop and signifying the end of program.
+     */
+    private static void setDone() {
+        done = true;
+    }
+
+    /**
+     * This function check if done statement and index in done statement is valid.
+     * @param input input given by the user.
+     * @return returns the validity of the done input.
+     */
+    private static boolean isDoneInput(String input) {
+        boolean validDoneStatement = Pattern.matches("^done \\d+$", input.toLowerCase());
+        if (!validDoneStatement) {
+            return false;
+        }
+        int numberOnly = Integer.parseInt(input.replaceAll("[^0-9]", ""));
+        return numberOnly <= listIndexTracker && numberOnly < 100;
```

Comment:
Here maybe can extract "numberOnly <= listIndexTracker" and "numberOnly < 100" to two boolean variables and it will be a bit easier to read.